programs/mango-v4/src/instructions/token_deposit.rs

```diff
@@ -140,10 +144,7 @@ impl<'a, 'info> DepositCommon<'a, 'info> {
         };
 
         // Transfer the actual tokens
-        token::transfer(
-            self.transfer_ctx(),
-            amount_i80f48.round_to_zero().to_num::<u64>(),
-        )?;
+        token::transfer(self.transfer_ctx(), amount_i80f48.to_num::<u64>())?;
```

Comment:
this to_num() is a bit unfortunate, but should always succeed. I think this might possibly be rewritten such that we have an integer amount here instead of the I80F48 (convert to integer near the .abs().ceil(), but even that is a bit awkward; I recently made a `I80F48::clamp_to_u64()` for situations like this in another branch)